lang/cs/Org.Apache.REEF.Experimental/ParquetReader/ParquetCollection.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.IO;
+using Microsoft.Hadoop.Avro.Container;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Org.Apache.REEF.Experimental.ParquetCollection
+{
+    /// <summary>
+    /// Represents a data collection of type T from parquet file.
+    /// </summary>
+    public class ParquetCollection<T> : IEnumerable<T>, IDisposable
+    {
+        private readonly Stream stream;
+        private readonly IAvroReader<T> avroReader;
+        private readonly SequentialReader<T> seqReader;
+        private readonly string _avroPath;
+        private bool _disposed = false;
+
+        /// <summary>
+        /// Constructor of ParquetCollection
+        /// </summary>
+        /// <param name="avroPath">Path to input avro file.</param>
+        public ParquetCollection(string avroPath)
```

Comment:
This constructor can throw all sorts of IO Exceptions. Maybe catch them all and wrap them, and then document which one it throws in the XML Doc?